### PR TITLE
CPDTP-520 trigger eligibility from eligibility status changes

### PR DESCRIPTION
**Pull request by blackrat**
## Ticket and context
When eligibility is checked and a profile is marked as eligible, this needs to cascade into any declarations that have been made for that profile (participant), across all event types. (i.e. if they have been declared as started, retained-1, etc.), their declarations will sit in the "submitted" state until they pass eligibility checks (lead provider training at risk). 

This service is now linked to the service that provides that update to "eligible" and once that triggers, it will update all submitted declarations attached to that profile.

Ticket: https://dfedigital.atlassian.net/browse/CPDTP-520

### Code quality checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### Gotchas
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.

## External API changes

Does this change anything in our external APIs? If so, did you remember to update the CHANGELOG for Lead Providers? (docs/source/release-notes)

### CODE REVIEW

File: app/services/store_participant_eligibility.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ def call
 
     update_and_store_eligibility_values!
 
+    RecordDeclarations::Actions::MakeDeclarationsEligibleForParticipantProfile.call(participant_profile: participant_profile) if changed_to_eligible?
```
Comment (kicferk1) [inline app/services/store_participant_eligibility.rb:18]: `refresh_payability!` has a similar check, right? Should the `MakeDeclarationsEligibleForParticipantProfile` job just call `refresh_payability` on the relevant declarations?

File: spec/services/record_declarations/actions/make_declarations_eligible_for_participant_profile_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  context "#call" do
+    it "starts with all declarations set to submitted" do
+      ParticipantDeclaration.all.each do |participant_declaration|
+        expect(participant_declaration.submitted?).to be_truthy
+      end
+    end
+
+    it "marks any submitted declarations for the participant as eligible" do
+      eligibility.eligible_status!
+      described_class.call(participant_profile: participant_profile)  # TODO: Remove this line when it's wired up automatically to the eligibility status check.
```
Comment (kicferk1) [inline spec/services/record_declarations/actions/make_declarations_eligible_for_participant_profile_spec.rb:20]: When can we address that todo?

File: app/services/store_participant_eligibility.rb (hunk 1 above)
Comment (blackrat) [reply app/services/store_participant_eligibility.rb]: Refresh payability isn't required any longer. It's removed by a later commit

File: spec/services/record_declarations/actions/make_declarations_eligible_for_participant_profile_spec.rb (hunk 2 above)
Comment (blackrat) [reply spec/services/record_declarations/actions/make_declarations_eligible_for_participant_profile_spec.rb]: Only a TODO for this draft.

File: app/services/record_declarations/base.rb [hunk 3]
```diff
@@ -65,13 +64,8 @@ def create_declaration_attempt!
...
     def find_or_create_record!
-      ApplicationRecord.transaction do
-        self.class.declaration_model.find_or_create_by!(declaration_parameters) do |participant_declaration|
-          DeclarationState.create!(
-            participant_declaration: participant_declaration,
-            state: user_profile.fundable? ? "eligible" : "submitted",
-          )
-        end
+      self.class.declaration_model.find_or_create_by!(declaration_parameters).tap do |participant_declaration|
+        participant_declaration.make_eligible! if fundable?
```
Comment (kicferk1) [inline app/services/record_declarations/base.rb:68]: The old code created `submitted` state. Does the new code do that too?

File: app/services/store_participant_eligibility.rb [hunk 4]
```diff
@@ -24,6 +18,17 @@ def call
...
+  attr_reader :participant_profile, :eligibility_options
+
+  def initialize(participant_profile:, eligibility_options: {})
+    @participant_profile = participant_profile
+    @eligibility_options = eligibility_options
+  end
+
+  def eligibility_triggers!
+    RecordDeclarations::Actions::MakeDeclarationsEligibleForParticipantProfile.call(participant_profile: participant_profile)
+  end
```
Comment (kicferk1) [inline app/services/store_participant_eligibility.rb:30]: Do we need to be able to go the other way? i.e when participant becomes ineligible, change all eligible declarations to submitted?

File: app/services/record_declarations/base.rb (hunk 3 above)
Comment (blackrat) [reply app/services/record_declarations/base.rb:68]: It's the default and now creates submitted first and then eligible, so it does have a definite first state.

**Pull request merged**